util/testutil/testing.go

```diff
@@ -62,9 +63,16 @@ func NotOk(tb TB, err error, format string, a ...interface{}) {
 }
 
 // Equals fails the test if exp is not equal to act.
-func Equals(tb TB, exp, act interface{}) {
+func Equals(tb TB, exp, act interface{}, msgAndArgs ...interface{}) {
 	tb.Helper()
 	if !reflect.DeepEqual(exp, act) {
-		tb.Fatalf("\033[31m\nexp: %#v\n\ngot: %#v\033[39m\n", exp, act)
+		tb.Fatalf("\033[31m\nexp: %#v\n\ngot: %#v%s\033[39m\n", exp, act, formatMessage(msgAndArgs))
 	}
 }
+
+func formatMessage(msgAndArgs []interface{}) string {
+	if len(msgAndArgs) == 0 {
+		return ""
+	}
+	return fmt.Sprintf(fmt.Sprintf("\n\nmsg: %s", msgAndArgs[0]), msgAndArgs[1:]...)
```

Comment:
This looks odd, why are you taking the first argument as a format string?

Reply:
The goal was to have an optional message that could have formatters in it, similar to the [tesitfy equal](https://github.com/stretchr/testify/blob/master/assert/assertions.go#L328).

I find that much of the time a message is not needed for an `Equals`, but sometimes you want to add the extra info, and passing an empty string everywhere that uses `Equals` is not a pattern I like.

If that pattern is ok, perhaps I should add a comment to Equals explaining how to format messages?